File: Solutions/1886. Determine Whether Matrix Can Be Obtained By Rotation/1886. Determine Whether Matrix Can Be Obtained By Rotation.cpp
class Solution {
public:
    bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
        bool a,b,c,d;
        a=b=c=d=true;
        int n=mat.size();
        for(int i=0;i<n;++i)
            for(int j=0;j<n;++j){
                // 0 degree
                if(mat[i][j]!=target[i][j])
                    a=false;
                // 90 degree
                if(mat[j][n-1-i]!=target[i][j])
                    b=false;
                // 180 degree
                if(mat[n-1-i][n-1-j]!=target[i][j])
                    c=false;
                // 270 degree
                if(mat[n-1-j][i]!=target[i][j])
                    d=false;
            }
        return a or b or c or d;
    }
};

File: Solutions/1886. Determine Whether Matrix Can Be Obtained By Rotation/1886. Determine Whether Matrix Can Be Obtained By Rotation.java
class Solution {
    public boolean findRotation(int[][] mat, int[][] target) {
        boolean[] c=new boolean[4];
        Arrays.fill(c,true);
        int n=mat.length;
        for(int i=0;i<n;++i)
            for(int j=0;j<n;++j){
                if(mat[i][j]!=target[i][j])
                    c[0]=false;
                if(mat[i][j]!=target[n-1-j][i])
                    c[1]=false;
                if(mat[i][j]!=target[j][n-1-i])
                    c[2]=false;
                if(mat[i][j]!=target[n-1-i][n-1-j])
                    c[3]=false;
            }
        return c[0] || c[1] || c[2] || c[3];
    }
}